docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+
+We highly recommend checking out our [Community Resources](#DOCS_TOPICS_COMMUNITY_RESOURCES/interactions) and the two libraries found there. They not only provide typing for Interactions data models, but also include decorators for API frameworks like Flask and Express to make validation easy.
+
+## Subcommands and Subcommand Groups
+
+> warn
+> Currently, subcommands and subcommand groups all appear at the top level in the command explorer. This may change in the future to include them as nested autocomplete options.
+
+For those developers looking to make more organized and complex groups of commands, look no further than subcommands and groups.
+
+**Subcommands** organize your command by **specifying actions within a command or group**.
+
+**Subcommand Groups** organize your **subcommands** by **grouping commands by similar action or resource within a command**.
+
+These are not enforced rules. You are free to use subcommands and groups however you'd like; it's just how we think about them.
+
+> danger
+> Using subcommands or subcommand groups will make your base command unusable. You can't access the base `/permissions` if you also have `/permissions add | remove` as subcommands or subcommand groups
+
+### Nested Subcommands and Groups
+
+A quick note on nested subcommands and groups. We support nesting one level deep within a group, meaning your top level command can contain subcommand groups, and those groups can contain subcommands. **That is the only kind of nesting supported.** Here's some visual examples:
+
+```
+VALID
+
+command
+|
+|__ subcommand-group
+    |
+    |__ subcommand
+|
+|__ subcommand-group
+    |
+    |__ subcommand
+
+
+-------
+
+INVALID
+
+
+command
+|
+|__ subcommand-group
+    |
+    |__ subcommand-group
+|
+|__ subcommand-group
+    |
+    |__ subcommand-group
+
+----
+
+INVALID
+
+command
+|
+|__ subcommand
+    |
+    |__ subcommand-group
+|
+|__ subcommand
+    |
+    |__ subcommand-group
+```
+
+### Example Walkthrough
+
+Let's look at an example. Let's imagine you run a moderation bot. You want to make a `/permissions` command that can do the following:
+
+- Get the guild permissions for a user or a role
+- Get the permissions for a user or a role on a specific channel
+- Change the guild permissions for a user or a role
+- Change the permissions for a user or a role on a specific channel
+
+We'll start by defining the top-level information for `/permissions`:
+
+```js
+{
+    "name": "permissions",
+    "description": "Get or edit permissions for a user or a role",
+    "options": []
+}
+```
+
+![A command with no arguments. It says /permissions](command.png)
+
+Now we have a command named `permissions`. We want this command to be able to affect users roles. Rather than making two separate commands, we can use subcommand groups. We want to use subcommand groups here because we are grouping commands on a similar resource: `user` or `role`.
```

Comment:
```suggestion
Now we have a command named `permissions`. We want this command to be able to affect users and roles. Rather than making two separate commands, we can use subcommand groups. We want to use subcommand groups here because we are grouping commands on a similar resource: `user` or `role`.
```